libraries/rush-lib/src/logic/operations/Operation.ts

```diff
@@ -23,6 +23,11 @@ export interface IOperationOptions {
    * running the operation.
    */
   runner?: IOperationRunner | undefined;
+
+  shard?: {
+    current: number;
+    total: number;
+  };
```

Comment:
I'm fine with promoting `outputFolderNames` to be a property of `Operation` if you need a place that is accessible to the `CacheableOperationPlugin`, but `Operation` has no business knowing anything about the concept of shards. That's a runner detail.

Reply:
Definitely, updated in [6c8cf5c](https://github.com/microsoft/rushstack/pull/4652/commits/6c8cf5ce262653513436729cb978a815179ea525)